src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -6,78 +6,184 @@
 
 namespace Microsoft.AspNetCore.Routing
 {
+    /// <summary>
+    /// Defines a contract to generate URLs to endpoints.
+    /// </summary>
     public abstract class LinkGenerator
     {
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <returns>The generated absolute URL.</returns>
         public string GetLink(object values)
         {
             return GetLink(httpContext: null, routeName: null, values, options: null);
         }
 
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="options">The <see cref="LinkOptions"/>.</param>
+        /// <returns>The generated absolute URL.</returns>
         public string GetLink(object values, LinkOptions options)
         {
             return GetLink(httpContext: null, routeName: null, values, options);
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="link">The generated absolute URL.</param>
+        /// <returns><c>true</c> if a URL was generated successfully, <c>false</c> otherwise.</returns>
         public bool TryGetLink(object values, out string link)
         {
             return TryGetLink(httpContext: null, routeName: null, values, options: null, out link);
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="options">The <see cref="LinkOptions"/>.</param>
+        /// <param name="link">The generated absolute URL.</param>
+        /// <returns><c>true</c> if a URL was generated successfully, <c>false</c> otherwise.</returns>
         public bool TryGetLink(object values, LinkOptions options, out string link)
         {
             return TryGetLink(httpContext: null, routeName: null, values, options, out link);
         }
 
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> associated with current request.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <returns>The generated absolute URL.</returns>
         public string GetLink(HttpContext httpContext, object values)
         {
             return GetLink(httpContext, routeName: null, values, options: null);
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
```

Comment:
I like the summary int.TryParse has for the TryParse pattern - https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=netframework-4.7.2

Just have the standard summary and add "A return value indicates whether the operation succeeded." to the end.

When it comes to XML docs I tend to copy what other people have already done 😄